### ISSUE TITLE
[Bug]: Nightly tests hang [standalone]

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#11148]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: dd24ea3
- Deployment mode(standalone or cluster): standalone
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

On standalone mode, nightly test cases (L0, L1, L2) are hanging.

Previously maximum dim and partition numbers test cases are skipped.

And in cluster mode, tests hangs for issue [11304](https://github.com/milvus-io/milvus/issues/11034).

While in standalone mode, it hangs. But from test client, it is no any errors which is different from cluster mode, so file another issue to track this in case of different root cause.

Logs:
standalone mode
https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/206/

### Expected Behavior

No tests hang

### Steps To Reproduce

```markdown
Run nightly tests
```

**Comment by binbinlv** [comment]
Block nightly standalone tests, so set it as urgent.

**Comment by binbinlv** [comment]
@congqixia  hello congqi,

could you help to check whether it is the same root cause of [11304](https://github.com/milvus-io/milvus/issues/11034)? thanks.

while I could not find any errors from test client which is different from [11304](https://github.com/milvus-io/milvus/issues/11034)

**Comment by congqixia** [comment]
from log there are some network related log. Look like same as #11034
```
default_milvus-testing-proxy-546fd45b-pmj8r_3e6773f4-f3e1-4765-bf6b-bf9564c74451/proxy/0.log
3978:2021-11-02T15:49:17.273002766Z stdout F [2021/11/02 15:49:17.272 +00:00] [DEBUG] [client.go:225] ["RootCoord Client grpc error"] [error="rpc error: code = Canceled desc = context canceled"]
3998:2021-11-02T15:49:17.274826203Z stdout F [2021/11/02 15:49:17.274 +00:00] [ERROR] [task_scheduler.go:453] ["Failed to execute task: rpc error: code = Canceled desc = context canceled"] [traceID=2505905b70f02bb5] [stack="github.com/milvus-io/milvus/internal/proxy.(*taskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:453\ngithub.com/milvus-io/milvus/internal/proxy.(*taskScheduler).definitionLoop\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:478"]
4003:2021-11-02T15:49:17.274865478Z stdout F [2021/11/02 15:49:17.274 +00:00] [DEBUG] [client.go:195] ["QueryCoord Client grpc error"] [error="rpc error: code = Canceled desc = context canceled"]
4022:2021-11-02T15:49:17.275897721Z stdout F [2021/11/02 15:49:17.275 +00:00] [ERROR] [task_scheduler.go:453] ["Failed to execute task: call query coordinator LoadCollection: rpc error: code = Canceled desc = context canceled"] [traceID=55d06fec1c071aee] [stack="github.com/milvus-io/milvus/internal/proxy.(*taskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:453\ngithub.com/milvus-io/milvus/internal/proxy.(*taskScheduler).definitionLoop\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:478"]
4061:2021-11-02T15:49:37.364228291Z stdout F [2021/11/02 15:49:37.364 +00:00] [DEBUG] [client.go:225] ["RootCoord Client grpc error"] [error="rpc error: code = Canceled desc = context canceled"]
4081:2021-11-02T15:49:37.365705902Z stdout F [2021/11/02 15:49:37.365 +00:00] [ERROR] [task_scheduler.go:453] ["Failed to execute task: rpc error: code = Canceled desc = context canceled"] [traceID=16add96da5ca5f68] [stack="github.com/milvus-io/milvus/internal/proxy.(*taskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:453\ngithub.com/milvus-io/milvus/internal/proxy.(*taskScheduler).definitionLoop\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:478"]
4102:2021-11-02T15:49:39.280552644Z stdout F [2021/11/02 15:49:39.280 +00:00] [DEBUG] [client.go:195] ["QueryCoord Client grpc error"] [error="rpc error: code = Canceled desc = context canceled"]
4121:2021-11-02T15:49:39.281759765Z stdout F [2021/11/02 15:49:39.281 +00:00] [ERROR] [task_scheduler.go:453] ["Failed to execute task: call query coordinator LoadCollection: rpc error: code = Canceled desc = context canceled"] [traceID=3bf97c1f2765a4db] [stack="github.com/milvus-io/milvus/internal/proxy.(*taskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:453\ngithub.com/milvus-io/milvus/internal/proxy.(*taskScheduler).definitionLoop\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:478"]
4150:2021-11-02T15:49:59.324496619Z stdout F [2021/11/02 15:49:59.324 +00:00] [DEBUG] [client.go:225] ["RootCoord Client grpc error"] [error="rpc error: code = Canceled desc = context canceled"]
4170:2021-11-02T15:49:59.325986702Z stdout F [2021/11/02 15:49:59.325 +00:00] [ERROR] [task_scheduler.go:453] ["Failed to execute task: rpc error: code = Canceled desc = context canceled"] [traceID=278f8b17405298a6] [stack="github.com/milvus-io/milvus/internal/proxy.(*taskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:453\ngithub.com/milvus-io/milvus/internal/proxy.(*taskScheduler).definitionLoop\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:478"]
4191:2021-11-02T15:50:01.284538819Z stdout F [2021/11/02 15:50:01.284 +00:00] [DEBUG] [client.go:195] ["QueryCoord Client grpc error"] [error="rpc error: code = Canceled desc = context canceled"]
4210:2021-11-02T15:50:01.285763831Z stdout F [2021/11/02 15:50:01.285 +00:00] [ERROR] [task_scheduler.go:453] ["Failed to execute task: call query coordinator LoadCollection: rpc error: code = Canceled desc = context canceled"] [traceID=5d17da8690d6c328] [stack="github.com/milvus-io/milvus/internal/proxy.(*taskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:453\ngithub.com/milvus-io/milvus/internal/proxy.(*taskScheduler).definitionLoop\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:478"]
4489:2021-11-02T15:50:23.290530594Z stdout F [2021/11/02 15:50:23.290 +00:00] [DEBUG] [client.go:195] ["QueryCoord Client grpc error"] [error="rpc error: code = Canceled desc = context canceled"]
4508:2021-11-02T15:50:23.291716132Z stdout F [2021/11/02 15:50:23.291 +00:00] [ERROR] [task_scheduler.go:453] ["Failed to execute task: call query coordinator LoadCollection: rpc error: code = Canceled desc = context canceled"] [traceID=601fa1a6ed33248e] [stack="github.com/milvus-io/milvus/internal/proxy.(*taskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:453\ngithub.com/milvus-io/milvus/internal/proxy.(*taskScheduler).definitionLoop\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:478"]
4562:2021-11-02T15:50:43.29808168Z stdout F [2021/11/02 15:50:43.297 +00:00] [DEBUG] [client.go:195] ["QueryCoord Client grpc error"] [error="rpc error: code = Canceled desc = context canceled"]
4581:2021-11-02T15:50:43.29942629Z stdout F [2021/11/02 15:50:43.299 +00:00] [ERROR] [task_scheduler.go:453] ["Failed to execute task: call query coordinator LoadCollection: rpc error: code = Canceled desc = context canceled"] [traceID=79e918db0aba3c30] [stack="github.com/milvus-io/milvus/internal/proxy.(*taskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:453\ngithub.com/milvus-io/milvus/internal/proxy.(*taskScheduler).definitionLoop\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:478"]
4607:2021-11-02T15:50:45.294579798Z stdout F [2021/11/02 15:50:45.294 +00:00] [DEBUG] [client.go:195] ["QueryCoord Client grpc error"] [error="rpc error: code = Canceled desc = context canceled"]
4626:2021-11-02T15:50:45.295809246Z stdout F [2021/11/02 15:50:45.295 +00:00] [ERROR] [task_scheduler.go:453] ["Failed to execute task: call query coordinator LoadCollection: rpc error: code = Canceled desc = context canceled"] [traceID=4b35ab23f3d3ede3] [stack="github.com/milvus-io/milvus/internal/proxy.(*taskScheduler).processTask\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:453\ngithub.com/milvus-io/milvus/internal/proxy.(*taskScheduler).definitionLoop\n\t/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:478"]

```

**Comment by binbinlv** [comment]
Lower down the parallel number from 6 to 4. And tests still hangs from test client side:
```
[2021-11-06T19:47:42.633Z] testcases/test_utility.py::TestUtilityBase::test_calc_distance_right_vectors_and_partition_ids[metric-L2-False] 

[2021-11-06T19:47:42.633Z] [gw2] [ 84%] PASSED testcases/entity/test_insert.py::TestInsertBinary::test_insert_binary_partition 

[2021-11-06T21:05:06.203Z] testcases/entity/test_insert.py::TestInsertBinary::test_insert_binary_multi_times timeout: sending signal TERM to command '/home/jenkins/agent/workspace/tests/scripts/e2e.sh'
```

https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/210/pipeline/71/

Logs:
https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/210/

**Comment by binbinlv** [comment]
And in cluster mode, tests hangs at the very beginning.
```
[2021-11-06T15:10:37.443Z] [gw2] [  8%] PASSED testcases/test_collection.py::TestGetCollectionStats::test_get_collection_stats_empty 
[2021-11-06T15:10:41.623Z] testcases/test_collection.py::TestGetCollectionStats::test_get_collection_stats_partition 
[2021-11-06T15:10:41.624Z] [gw3] [  8%] PASSED testcases/test_collection.py::TestCollectionMultiCollections::test_collection_count_multi_collections_l2[1000] 
[2021-11-07T01:08:03.882Z] testcases/test_collection.py::TestCollectionMultiCollections::test_collection_count_multi_collections_binary[2001] timeout: sending signal TERM to command '/home/jenkins/agent/workspace/tests/scripts/e2e.sh'
[2021-11-07T01:08:03.882Z] + cleanup_kind_cluster kind
[2021-11-07T01:08:03.882Z] + echo 'Test exited with exit code 124.'
[2021-11-07T01:08:03.882Z] Test exited with exit code 124.
```

https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/210/pipeline

Logs:
https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/210/

**Comment by binbinlv** [comment]
The latest nightly hangs again:

https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/212/pipeline

Logs:
https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/212/

**Comment by binbinlv** [comment]
@congqixia any updates?

And could you please update what you have found in this issue?  Thanks.

**Comment by congqixia** [comment]
> @congqixia any updates?
> 
> And could you please update what you have found in this issue? Thanks.

Nightly 210 and 212 both hang when datanode tried to Seek using MqttMsgstream. In distributed cases, the under-hood component should be pulsar. Still trying to find the root cause.

**Comment by binbinlv** [comment]
Nightly 213 still exists;

https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/213/pipeline

Logs:
https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/213/

Timeline:
```
[2021-11-09T15:26:47.228Z] [gw0] [  9%] PASSED testcases/test_collection.py::TestGetCollectionStats::test_get_collection_stats_after_index_created[get_simple_index5] 

[2021-11-10T01:22:20.033Z] testcases/test_collection.py::TestGetCollectionStats::test_get_collection_stats_after_index_created_ip[get_simple_index0] timeout: sending signal TERM to command '/home/jenkins/agent/workspace/tests/scripts/e2e.sh'
```

**Comment by binbinlv** [comment]
Latest nightly with similar issue:

https://ci.milvus.io:18080/jenkins/blue/rest/organizations/jenkins/pipelines/milvus-nightly-ci/branches/master/runs/215/nodes/71/steps/88/log/?start=0

```
[2021-11-11T19:33:22.073Z] [gw3] [ 91%] PASSED testcases/test_search_20.py::TestSearchBase::test_search_index_empty_partition[get_simple_index0-10-1100] 
[2021-11-11T19:33:36.943Z] testcases/test_search_20.py::TestSearchBase::test_search_index_empty_partition[get_simple_index1-1-1] 
[2021-11-11T22:23:35.164Z] [gw3] [ 91%] PASSED testcases/test_search_20.py::TestSearchBase::test_search_index_empty_partition[get_simple_index1-1-1] timeout: sending signal TERM to command '/home/jenkins/agent/workspace/tests/scripts/e2e.sh'
```

Logs:
https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/215/

**Comment by binbinlv** [comment]
Continuously hanging:
Latest running:

Logs:
https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/218/

Standalone:
```
[2021-11-14T18:00:15.908Z] testcases/test_search_20.py::TestSearchBase::test_search_index_empty_partition[get_simple_index1-1-1] 

[2021-11-14T21:02:43.817Z] [gw3] [ 91%] PASSED testcases/test_search_20.py::TestSearchBase::test_search_index_empty_partition[get_simple_index1-1-1] timeout: sending signal TERM to command '/home/jenkins/agent/workspace/tests/scripts/e2e.sh'
```

Cluster:
```
[2021-11-14T17:20:26.048Z] [gw2] [ 65%] PASSED testcases/test_search_20.py::TestSearchBase::test_search_ip_flat[get_simple_index7-10-1100] 

[2021-11-14T17:20:28.564Z] testcases/test_search_20.py::TestCollectionSearch::test_search_multi_collections[3000-2-128-False-False] 

[2021-11-15T01:08:46.759Z] testcases/test_search_20.py::TestSearchBase::test_search_ip_flat[get_simple_index8-1-1] timeout: sending signal TERM to command '/home/jenkins/agent/workspace/tests/scripts/e2e.sh'
```

**Comment by binbinlv** [comment]
Continuously hanging: Latest running:

standalone mode:

Logs: [artifacts-milvus-standalone-master-220-pymilvus-e2e-logs.tar.gz]

https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/220/artifacts

hanging timeline:
```
[2021-11-16T15:37:27.307Z] [gw4] [ 41%] FAILED testcases/test_search_20.py::TestCollectionSearch::test_search_after_different_index_with_params[8-False-True-RHNSW_SQ-params8] 
[2021-11-16T21:08:56.685Z] [gw5] [ 41%] FAILED testcases/test_search_20.py::TestCollectionSearch::test_search_collection_after_release_load[3000-500-128-False-False] 
```

**Comment by binbinlv** [comment]
Recently, for standalone mode, tests do not hang, so cancel the urgent label first. But it is still fails, so did not confirm this issue is covered or fixed. so still keep this issue open.

**Comment by binbinlv** [comment]
Set for urgent, as it is blocked for hang issue again.

**Comment by binbinlv** [comment]
Latest nightly hangs issue:

https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/231/pipeline/91

Logs:
https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/231/artifacts

**Comment by xige-16** [comment]
> Latest nightly hangs issue:
> 
> https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/231/pipeline/91
> 
> Logs: https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/231/artifacts

standalone panic when flush

**Comment by xige-16** [comment]
blocked by #12265

**Comment by binbinlv** [comment]
Latest nightly: (standalone)

milvus: 2c484c7
pymilvus: 2.0.0rc9.dev16

Logs:
https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/234/

**Comment by xige-16** [comment]
[2021-11-28T18:20:12.132Z] testcases/test_utility.py::TestUtilityBase::test_calc_distance_left_vector_and_collection_ids[metric-IP-True] 
[2021-11-28T18:20:14.655Z] [gw2] [ 90%] PASSED testcases/test_utility.py::TestUtilityBase::test_calc_distance_left_vector_and_collection_ids[metric-IP-True] 
[2021-11-28T18:20:21.210Z] testcases/test_utility.py::TestUtilityBase::test_calc_distance_left_vector_and_collection_ids[metric-IP-False] 
[2021-11-29T01:50:51.613Z] [gw2] [ 91%] PASSED testcases/test_utility.py::TestUtilityBase::test_calc_distance_left_vector_and_collection_ids[metric-IP-False]

**Comment by xige-16** [comment]
> [2021-11-28T18:20:12.132Z] testcases/test_utility.py::TestUtilityBase::test_calc_distance_left_vector_and_collection_ids[metric-IP-True] [2021-11-28T18:20:14.655Z] [gw2] [ 90%] PASSED testcases/test_utility.py::TestUtilityBase::test_calc_distance_left_vector_and_collection_ids[metric-IP-True] [2021-11-28T18:20:21.210Z] testcases/test_utility.py::TestUtilityBase::test_calc_distance_left_vector_and_collection_ids[metric-IP-False] [2021-11-29T01:50:51.613Z] [gw2] [ 91%] PASSED testcases/test_utility.py::TestUtilityBase::test_calc_distance_left_vector_and_collection_ids[metric-IP-False]

ci block here

**Comment by DragonDriver** [comment]
> [2021-11-28T18:20:12.132Z] testcases/test_utility.py::TestUtilityBase::test_calc_distance_left_vector_and_collection_ids[metric-IP-True] [2021-11-28T18:20:14.655Z] [gw2] [ 90%] PASSED testcases/test_utility.py::TestUtilityBase::test_calc_distance_left_vector_and_collection_ids[metric-IP-True] [2021-11-28T18:20:21.210Z] testcases/test_utility.py::TestUtilityBase::test_calc_distance_left_vector_and_collection_ids[metric-IP-False] [2021-11-29T01:50:51.613Z] [gw2] [ 91%] PASSED testcases/test_utility.py::TestUtilityBase::test_calc_distance_left_vector_and_collection_ids[metric-IP-False]

I add some logs for this rpc in #12371 . I hope that will help us debug more easier.

**Comment by binbinlv** [comment]
Now through changing the searching behavior during releasing from returning empty to reporting message, now the tests are not hanging, so close it.
